packages/react-i18n/src/index.tsx

```diff
@@ -65,12 +71,19 @@ export const withI18n = createHigherOrderComponent< I18nReact >( ( InnerComponen
  * Utility to make a new context value
  *
  * @param localeData The localeData
+ * @param localeSlug Locale slug to use instead of the one found in `localeData`
  *
  * @returns The context value with bound translation functions
  */
-function makeContextValue( localeData?: LocaleData ): I18nReact {
+function makeContextValue( localeData?: LocaleData, localeSlug?: string ): I18nReact {
+	const i18nLocale = localeSlug ?? localeData?.[ '' ]?.localeSlug ?? 'en';
```

Comment:
Neither accessing localeSlug or the `en` fallback work here, localeSlug is non standard and en fallback can't be assumed otherwise.

Reply:
(the en fallback above is fine, I just don't think it makes sense in this package)

Reply:
Indeed. The `useI18n()` hook guarantees that it'll always return a value for `i18nLocale`. However maybe that's not realistic. Perhaps we should change the return type of `useI18n()` to this:

```typescript
export interface I18nReact {
	__: I18n[ '__' ];
	_n: I18n[ '_n' ];
	_nx: I18n[ '_nx' ];
	_x: I18n[ '_x' ];
	isRTL: I18n[ 'isRTL' ];
	i18nLocale?: string;
}
```

Note the **?** on the `i18nLocale` prop. That way we'd fall back to `undefined` instead of `'en'`.

**Another** option I've been thinking about is completely removing the `i18nLocale` prop from `useI18n()`. `react-i18n` is supposed to be something we could push upstream, and since having the locale slug available in the locale data is non-standard, it probably won't survive the review process.

We've been explicitly referring to the locale slug a lot in gutenboarding, but that's a special case because the user is logged out, and they can provide the locale in the url. That's not usually how things work. Perhaps the locale slug can be provided to a *new* thing, like a `<LocaleProvider>`, that's part of an a8c-specific. That way we can depend on it for our a8c helper functions like `localizeUrl()` and we know we can provide it at app boot time because of the idiosyncrasies of our WordPress environment.

Reply:
I don't think an optional locale slug value works - users of the prop will have to deal with that case everywhere its used, the default value needs to be decided by the consumer of react-i18n. In a lot of cases that will be english but its an assumption we can't force on everyone.

This another other option sounds really good - feels much cleaner and I like stepping away from making decisions about what a locale is.

Reply:
#47446 - I like this solution to break apart the locale slug from the locale data, but we need some changes to break away from `calypso-i18n` https://github.com/Automattic/wp-calypso/pull/47446#pullrequestreview-530931752

cc @Automattic/i18n might want to be aware of this

Reply:
I think we should go with #47446. It probably needs to merge first so this one can be updated without breaking the build.